routes/api/contacts.js

```diff
@@ -1,25 +1,113 @@
+    console.log(contacts);
+    res.status(200).json(contacts);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Unknown error occurred" });
+  }
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:id", async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    const contact = await contactsMethots.getContactById(id);
+    if (!contact) {
+      return res.status(404).json({ message: "Contact not found" });
+    }
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+    res.status(200).json(contact);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Unknown error occurred" });
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const validation = schema.validate(req.body);
+    if (validation.error) {
+      res.status(400).json({ error: validation.error.message });
+      return;
+    }
+    const contactsAdded = await contactsMethots.addContact(req.body);
+    res.status(201).json(contactsAdded);
+  } catch (error) {
+    console.error("Error adding contact:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.delete("/:contactId", async (req, res, next) => {
+  const contactId = req.params.contactId;
+  try {
+    const contact = await contactsMethots.getContactById(contactId);
+    if (!contact) {
+      return res.status(404).json({ error: "Contact not found" });
+    }
+    await contactsMethots.removeContact(contactId);
+    res.status(200).json({ message: "Contact deleted" });
+  } catch (error) {
+    console.error("Error deleting contact:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
 
-module.exports = router
+router.put("/:contactId", async (req, res, next) => {
+  const contactId = req.params.contactId;
+  const body = req.body;
+  try {
+    const validation = schema.validate(req.body);
+    if (validation.error) {
+      res.status(400).json({ error: validation.error.message });
+    }
+    const contact = await contactsMethots.getContactById(contactId);
+    if (!contact) {
+      return res.status(404).json({ error: "Contact not found" });
+    }
+
+    await contactsMethots.updateContact(contactId, body);
+    res.status(200).json({ message: "Contact updated" });
+  } catch (error) {
+    console.error("Error deleting contact:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+router.patch("/:contactId/favorite", async (req, res, next) => {
+  const contactId = req.params.contactId;
+  const body = req.body.favorite;
+  try {
+    const validation = schema.validate(req.body);
```

Comment:
Powinnaś mieć 3 osobne schemy dla post put i patch. Czemu tutaj muszę podawać name i email, to trochę nie ma sensu. To samo w put raczej nie musi być w każdym polu required() bo mogę chcieć edytować np tylko email